### ISSUE TITLE
[BUG][CV] deepchecks should either validate model.eval status or actively create it 

### ISSUE THREAD

**Issue by noamzbr** [deepchecks/deepchecks#1030]
**Describe the bug**
Checks can crash or give wrong results if models are set by mistake to training mode. 

**To Reproduce**
Run metrics check with model.train() preceding it.

**Expected behavior**
Either we validate that the model is in the correct state, or we actively call model.eval()
